test/jasmine/tests/calcdata_test.js

```diff
@@ -28,4 +29,215 @@ describe('calculated data and points', function() {
             expect(gd.calcdata[0][3]).toEqual({ x: false, y: false});
         });
     });
+
+    xdescribe('category ordering', function() {
+
+        describe('default category ordering reified', function() {
+
+            it('should output categories in the given order by default', function() {
+
+                Plotly.plot(gd, [{x: ['c','a','e','b','d'], y: [15,11,12,13,14]}], { xaxis: {
+                    type: 'category'
+                }});
+
+                expect(gd.calcdata[0][0].y).toEqual(15);
+                expect(gd.calcdata[0][1].y).toEqual(11);
+                expect(gd.calcdata[0][2].y).toEqual(12);
+                expect(gd.calcdata[0][3].y).toEqual(13);
+                expect(gd.calcdata[0][4].y).toEqual(14);
+            });
+
+            it('should output categories in the given order if `trace` order is explicitly specified', function() {
+
+                Plotly.plot(gd, [{x: ['c','a','e','b','d'], y: [15,11,12,13,14]}], { xaxis: {
+                    type: 'category',
+                    categorymode: 'trace'
+                    // Wouldn't it be preferred to supply a function and plotly would have several functions like this?
```

Comment:
it would be nice :smirk: , but only for JS users. 

All plotly.js attributes must be JSON serializable so that folks using our [python](https://github.com/plotly/plotly.py) and [R](https://github.com/ropensci/plotly) library can have access to the same features.